### PR TITLE
Fix the issue of changing setuid and setgid.

### PR DESCRIPTION
**Pull request by aabouzaid**
### What does this PR do?
The default behavior of Python `os.chown` (which used by the underlying function `chown`) does reset `setuid` and `setgid`.
And as descried in the issue, setting setuid/setgid only works on second run because of that.

I did test that with Py 2.6, 2.7, 3.
So setting the mode again after chown is required to fix this behavior.

### What issues does this PR fix or reference?
Issue no. #19834

The actual reason is that in the first time it changes the owner and group and resets the setuid/setgid, but in second time the owner/group are already changed, so `os.chown` is not invoked.

### Tests written?
Yes

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by terminalmage — COMMENTED** [review]
I like this. I think we should get an integration test added to make sure this doesn't regress, even though the code is well-commented.

**Comment by AAbouZaid** [comment]
@terminalmage I'd like to do it :-)
But since I'm not used to how it's done in SaltStack, it'd be great to give a good example to follow.

**Comment by rallytime** [comment]
Hi @AAbouZaid - Here's some docs on [getting started](https://docs.saltstack.com/en/latest/topics/tutorials/writing_tests.html) with Salt's test suite, here's some docs on [writing integration tests](https://docs.saltstack.com/en/latest/topics/development/tests/integration.html), and there are some examples in of tests already written in `salt.tests.integration.modules.test_file.py`. Let us know if you have any questions!

**Comment by AAbouZaid** [comment]
@rallytime Thanks for the links :-)
@terminalmage I'm still believe that this Jenkins setup is confusing, but it looks ready to go now :-)

File: salt/modules/file.py [hunk 1]
```diff
@@ -4453,6 +4453,9 @@ def check_perms(name, ret, user, group, mode, attrs=None, follow_symlinks=False)
                 group = perms['lgroup']
             try:
                 chown_func(name, user, group)
+                # Python os.chown() does rest the suid and sgid,
```
Comment (rrroo) [inline salt/modules/file.py]: Typo: `rest` = `reset`?

**Review by ghost — APPROVED** [review]
Thanks @AAbouZaid 
I'm looking forward to see this merged.

File: salt/modules/file.py (hunk 1 above)
Comment (AAbouZaid) [reply salt/modules/file.py]: Yes, it is :D 
I will fix it, thanks @rrroo :+1:

Comment (AAbouZaid) [reply salt/modules/file.py]: Fixed.

**Review by terminalmage — APPROVED** [review]
Test looks good, this is good to merge pending test results.

**Comment by rallytime** [comment]
@AAbouZaid There is a related test failing here: https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu16-py3/5527/testReport/junit/integration.states.test_file/FileTest/test_owner_after_setuid/

**Comment by AAbouZaid** [comment]
Fixed :-)

**Comment by rallytime** [comment]
re-run py

**Pull request merged**